packages/server/modules/core/repositories/objects.ts

```diff
@@ -249,3 +250,254 @@ export const getObjectChildrenFactory =
     const lastId = rows[rows.length - 1].id
     return { objects: rows, cursor: lastId }
   }
+
+/**
+ * This query is inefficient on larger sets (n * 10k objects) as we need to return the total count on an arbitrarily (user) defined selection of objects.
+ * A possible future optimisation route would be to cache the total count of a query (as objects are immutable, it will not change) on a first run, and, if found on a subsequent round, do a simpler query and merge the total count result.
+ */
+export const getObjectChildrenQueryFactory =
```

Comment:
yea, its massive, I tried to change as little as possible - the only parts I did change I had to because of TS